### PR TITLE
DDF-5606 Standardized references to 'resources' in documentation

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

Changes, where possible, references to 'data products' to the more accurate 'data resources' or simply 'resources'

#### Select relevant component teams: 
@codice/data 
@codice/docs 

#### Ask 2 committers to review/merge the PR and tag them here.
@ahoffer
@brendan-hofmann
@brjeter
@clockard
@lessarderic
@mcalcote
@pklinef
@ricklarsen - Documentation
@shaundmorris

#### How should this be tested?

review content

#### Any background context you want to provide?

Tried not to change references where the actual code being referenced uses the term 'product' instead of 'resource'

#### What are the relevant tickets?
Fixes: #5606 

#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-framework-intro.adoc [hunk 1]
```diff
@@ -165,8 +165,8 @@ It is possible to have only remote Sources configured and no local `CatalogProvi
 
 ===== Product Retrieval
```
Comment (jordanwilking) [inline distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-framework-intro.adoc]: ✏️ I noticed `Product Caching` seems to be different, but I wasn't sure about these (166, 171, 172)

File: distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-api.adoc [hunk 2]
```diff
@@ -101,30 +100,16 @@ Resources may exist either locally or on a remote data store.
...
+The Endpoint bundle contains a Web service that exposes the interface to retrieve resources.
 The Endpoint calls the `CatalogFramework` to execute the operations of its specification.
-The `CatalogFramework` relies on the Sources to execute the actual product retrieval.
-Optional PreResource and PostResource Catalog Plugins may be invoked by the `CatalogFramework` to modify the product retrieval request/response prior to the Catalog Provider processing the request and providing the response.
-It is possible to retrieve products from specific remote Sources by specifying the site name(s) in the request.
+The `CatalogFramework` relies on the Sources to execute the actual resource retrieval.
+Optional `PreResource` and `PostResource` Catalog Plugins may be invoked by the `CatalogFramework` to modify the resource retrieval request/response prior to the Catalog Provider processing the request and providing the response.
+It is possible to retrieve resources from specific remote Sources by specifying the site name(s) in the request.
 
 .Product Caching
```
Comment (bakejeyner) [inline distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-api.adoc]: ❓ Should all of the "Product" headers (Product Retrieval, Product Retrieval Services, Product Cacheing, etc.) be changed to "Resource" as well?

File: distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-api.adoc [hunk 3]
```diff
@@ -101,30 +100,16 @@ Resources may exist either locally or on a remote data store.
...
-====
-
-To configure product caching:
-
-. Navigate to the *${admin-console}*.
-. Select ${ddf-catalog}.
-. Select *Configuration*.
-. Select *Resource Download Settings*.
-
-See <<{reference-prefix}ddf.catalog.resource.download.ReliableResourceDownloadManager,Resource Download Settings configurations>> for all possible configurations.
```
Comment (bakejeyner) [inline distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-api.adoc]: ❓ Should we include a link to https://github.com/codice/ddf/pull/5608/files#diff-de98e1cea538c46c6ad3f393cd560bd7R1?

File: distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-framework-intro.adoc (hunk 1 above)
Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-framework-intro.adoc]: I didn't change any of the configurations (metatypes, for example) under the scope of this PR and they use "Product"

File: distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-api.adoc (hunk 2 above)
Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-api.adoc]: The configurations for these in the code use Product, so I left them for now, but it'd be nice to change them up as well

File: distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-api.adoc (hunk 3 above)
Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_architectures/_catalogFrameworks/catalog-api.adoc]: I'm not sure. Can you explain?

**Comment by ricklarsen** [comment]
build now

**Pull request merged**